src/protocols/rdp/channels/audio-input/audio-buffer.h

```diff
@@ -40,8 +51,8 @@
  *     The arbitrary data pointer provided when the audio buffer was
  *     initialized.
  */
-typedef void guac_rdp_audio_buffer_flush_handler(char* buffer, int length,
-        void* data);
+typedef void guac_rdp_audio_buffer_flush_handler(guac_rdp_audio_buffer* audio_buffer,
+        char* buffer, int length, void* data);
```

Comment:
Is there reason to expect that there are cases when `buffer` may be outside of the `audio_buffer`? Passing pointers around isn't that expensive, but seems like if we have the entire `audio_buffer` structure we could just retrieve `buffer` from there?

Reply:
We could, sure. The same could go for `data`, which is also part of `guac_rdp_audio_buffer`. Would you prefer something like:

```c
typedef void guac_rdp_audio_buffer_flush_handler(guac_rdp_audio_buffer* audio_buffer, int length);
```

?

Reply:
Yeah, that's fine with me - I wasn't sure if there were any situations where the function would be used to process a `buffer` and `data` that wasn't already part of `audio_buffer`? If not, I'm totally good with the above - if so, we can leave it as is. No strong preference...

Reply:
OK - I've removed the redundant parameters.